### ISSUE TITLE
List of potential ideas for exercises

### ISSUE THREAD

**Issue by AnshulMalik** [workshopper/learnrust#1]
Let's discuss here, what we can think of as interesting exercises with rust.

- Concept of mutable and non-mutable

You can take help from 
- https://doc.rust-lang.org/book/second-edition/ch01-00-introduction.html
- https://rust-lang-nursery.github.io/rust-cookbook/intro.html

Current Ideas:
- Print the arguments passed to the program

**Comment by Jay9596** [comment]
This list may look like the it's taking things from the links you provided and that may be true.
1. Basic
   - Hello World
   - Fibonacci / Factorial (both iterative and recursive ), helps with I/O (can use args), syntax for functions and different loops
   - Lucky 7 game ( a revised guessing game from the Rust book, useful to leach `match` syntax and using `random` crate
2. Some OOP style concepts using structs
3. String manipulation, uppercase, lowercase and maybe something like caesar cipher to help with `String` and `&str`
4. Echo HTTP server
5. File server / API endpoint
6. Multi-threading, (got nothing here..)
7. Start making some creates at this point.

These are my ideas on how a new Rust user can get themselves familiarized with the language.
What do you say ?

**Comment by AnshulMalik** [comment]
Sounds great! We can start looking for implementations, And I think @douglasduteil might have already started something ?

**Comment by Jay9596** [comment]
I would love to help with the implementation as well, please keep me in the loop about the progress and on how I can help.

**Comment by AnshulMalik** [comment]
Yep, will do, you can also drop https://gitter.im/workshopper/learnrust to chat

**Comment by douglasduteil** [comment]
<img width=693 src=https://user-images.githubusercontent.com/730511/34370350-a1d0dbba-eac3-11e7-9312-cd64ced8550a.gif >

---

I'm just a Rust Noob trying to learn Rust by coding Learn Rust 
@booyaa told me I could do it, so I'm trying :relaxed:

Last time I was practicing, I successfully build [a directory of exercises](https://github.com/douglasduteil/learnrust/tree/master/exercises) to a cursive UI. 

![screenshot from 2017-12-27 05-09-11](https://user-images.githubusercontent.com/730511/34370474-a358375c-eac4-11e7-8c73-8888d1d4432f.png)

I'm struggling with having this little code [built on multiple platforms](https://travis-ci.org/douglasduteil/learnrust/builds/301218011) I will have to look closer how [rustup](https://github.com/rust-lang-nursery/rustup.rs/blob/master/.travis.yml) or [cargo](https://github.com/rust-lang/cargo/blob/master/.travis.yml) manage this

**Comment by AnshulMalik** [comment]
You are doing great @douglasduteil I'll soon start working on this once I find time :)
